### ISSUE TITLE
Allow lhs of field assignment statement to be a macro invocation

### ISSUE THREAD

**Issue by fpoli** [fpoli/viper-silver#174]
This doesn't seem to parse, but it seems unclear what the problem is, looking at both the parse AST and eventual AST structures.

e.g.

```
#!scala

domain Array {
  function loc(a: Array, i: Int): Ref
  function len(a: Array): Int
  function first(r: Ref): Array
  function second(r: Ref): Int

  axiom all_diff {
    forall a: Array, i: Int :: {loc(a, i)}
      first(loc(a, i)) == a && second(loc(a, i)) == i
  }

  axiom length_nonneg {
    forall a: Array :: len(a) >= 0
  }
}

field val : Int

define A(i) loc(a,i).val

method test(a:Array) 
  requires acc(A(0)) // works
{ 
  var x:Int := A(0) // works
  A(0) := 0 // doesn't parse
}
```

[Issue created by alexander_summers: 2017-08-18]
[Last updated on bitbucket: 2019-02-01]

[Comment created by mschwerhoff: 2017-09-24]
See also issues #177, #178, #200

[Comment created by fabiopakk: 2019-02-01]
This issue was fixed in the following branches:

* fabiopakk_issue_177_macro_on_lhs_of_assignment
* fabiopakk_issue_178_field_deref_in_macros
* fabiopakk_issue_214_macro_on_lhs_of_assignment

All of them merged into default branch.

**Comment by fpoli** [comment]
> **@alexander_summers** commented on 2016-10-19 10:46

See also issue #53

**Comment by fpoli** [comment]
> **@mschwerhoff** commented on 2017-08-25 14:44

Resolved by @mousam05 with pull request #73
